testER/src/components/TrackButton.jsx

```diff
@@ -1,25 +1,24 @@
-import React, { useState } from 'react'
```

Comment:
A few notes on the changes here:

1. It now consumes our TrackContext. This is just an example of consolidated state management, you could use Redux or some other means if you prefer.
2. The way state in here was managed originally (`vis` and `tracks`) opens up surface area for bugs because it adds a layer of redundancy.  To explain a bit more:

In the original code, the initialization value of `vis` is `buttonTrack[subject.id]`.  This works because it gives you a boolean value which evaluates whether or not the subject is part of the track state. However, `toggleTracks` simply reverses that boolean value on click, meaning _instead of consistently being an evaluation of the subject's presence in the track state, it is now a black-boxed assumption about what is happening elsewhere in the app_.  This is why there were false on/off states in the popup when you were attempting to debug why the track state was not updating appropriately. The same approach was being applied for `setTracks`. This violates the idea of maintaining a single source of truth for your application's state.

As you can see now, we've absolved ourselves of state entirely because `vis` and `tracks` (renamed `imgSrc` for clarity) are just derived from our track state as such:

```
const vis = !!tracks[subject.id]
const imgSrc = vis ? '/public/images/button_icons/pin_tracks-green.png' : '/public/images/button_icons/pin_tracks-gray.png'
```
"visible === the track is turned on in state"
"our image source is green if visible, and gray otherwise"

No state management needed.